server/issue.go

```diff
@@ -193,6 +193,78 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
 	return http.StatusOK, nil
 }
 
+func (p *Plugin) assignJiraIssue(mmUserId, issueKey, jiraAssignee string) (*model.CommandResponse, error) {
```

Comment:
Why return both a response and an err?

Reply:
My fault -- I suggested that, since you need to provide the positive response... But I see now that you can provide the positive response in the command handler, and you only need to return errors from this function.

Reply:
@cpoile How do I provide the non-error type response if I only return an error type and not the *model.CommandResponse?

Reply:
@jfrerich well, you know a return of nil will be successful, so something like:
```
err = assignJiraIssue(...)
if err == nil {
  // do your error handling
}
return responsef(return the successful feedback here)
```

Reply:
model.CommandResponse does not provide for a separate stdin vs stderr, 0/5 do the simplest thing? (return a `string`? now where would that fit in an OO book?)

Reply:
@cpoile assignJiraIssue(...) returns the following if successful.  I return information gained from the function.  That is why I also have the *model.CommandResponse returned.  Otherwise, I would have just checked err == nil, like you said above.  Does that explanation make sense?

`	msg := fmt.Sprintf("`%s` assigned to Jira issue [%s](%s)", assignee.DisplayName, issueKey, permalink)
	return responsef(msg), nil

`

Reply:
@jfrerich gotcha, I misread assignee as something you were passing in. :) I think returning a (string, error) would probably work then, and put the string in a responsef in the caller. Keep the function as pure as possible (it doesn't need to know its result will be sent in a response).

Reply:
Thanks. I'm settling on (string, error) and will have the caller wrap the string in a responsef().